File: Kalk/model/aritmeticresult.cpp
#include "aritmeticresult.h"

AritmeticResult::AritmeticResult(const double ris) : Result(), ris(ris) {}

AritmeticResult::AritmeticResult(const AritmeticResult& e) : ris(e.ris) {}

AritmeticResult& AritmeticResult::operator =(const AritmeticResult& e)
{
    if(this != &e){
        ris = e.ris;
    }
    return *this;
}

AritmeticResult::operator double() const
{
    return ris;
}

double AritmeticResult::getValue() const
{
    return ris;
}

void AritmeticResult::stampa() const
{
    std::cout << "Ris: " << ris << std::endl;
}

File: KalkJava/Kalk/AritmeticResult.java
package Kalk;

public class AritmeticResult extends Result {
	private double ris;
	public AritmeticResult(double ris) {
		super();
		this.ris = ris;
	}
	public AritmeticResult(AritmeticResult e) {
		this.ris = e.ris;
	}

	@Override
	public final void stampa() {
		System.out.print("Ris: ");
		System.out.print(ris);
		System.out.print("\n");
	}

	public final double getValue() {
		return ris;
	}

	public final AritmeticResult copyFrom(AritmeticResult e) {
		if(this != e) {
			ris = e.ris;
		}
		return this;
	}

	public double toDouble() {
		return ris;
	}
}